model_analyzer/triton/model/model_config_variant.py

```diff
@@ -0,0 +1,29 @@
+# Copyright 2023, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from dataclasses import dataclass
+from typing import List
+
+from model_analyzer.triton.model.model_config import ModelConfig
+
+
+@dataclass
+class ModelConfigVariant:
+    """
+    A dataclass that holds the ModelConfig as well as the variant name
+    for the model
+    """
+
+    model_config: ModelConfig
+    variant_name: str
```

Comment:
Is there a reason we didn't just add variant_name into ModelConfig?

Reply:
Yes, this makes the code cleaner as a ModelConfig should not contain things (like the variant name) that are not part of the Triton model's metadata.

Reply:
Now everyone that wants anything from the model config has to do model_variant.model_config instead of just using the config. That seems less clean, that users need to know about this hierarchy.

Reply:
>Yes, this makes the code cleaner as a ModelConfig should not contain things (like the variant name) that are not part of the Triton model's metadata.

In this case, are we also considering to just make `ModelRunConfig` hold the extra information? It does seem like this change could introduce extra layer of calls/access.

Reply:
A good thought (and how I was originally coding it), but that's ugly in different way, as now a bunch of methods would have to pass back tuples of ModelConfig, variant_name, which is what I'm trying to avoid with this new dataclass